src/plugins/plugins_ex.h

```diff
@@ -386,119 +333,133 @@ struct breakpoint_by_pid_searcher
     }
 };
 
+template<typename Plugin, typename Params = void>
+struct plugin_data
+{
+    Plugin* plugin;
+    Params* params;
+
+    plugin_data(Plugin* plugin, Params* params)
+        : plugin(plugin), params(params)
+    {};
+};
+
 class pluginex : public plugin
 {
 public:
     typedef event_response_t(*hook_cb_t)(drakvuf_t drakvuf, drakvuf_trap_info_t* info);
 
-public:
-    pluginex(drakvuf_t drakvuf, output_format_t output);
-    virtual ~pluginex();
+    pluginex(drakvuf_t drakvuf, output_format_t output)
+        : m_output_format(output), drakvuf(drakvuf)
+    {};
 
-public:
-    template<typename P = pluginex, typename Params = plugin_params<P>, typename IB,
-             typename AP = allocator<Params>, typename AT = allocator<drakvuf_trap_t>>
-    drakvuf_trap_t* register_trap(drakvuf_t drakvuf,
-                                  drakvuf_trap_info_t* info,
-                                  P* plugin,
+    virtual ~pluginex()
+    {};
+
+    // Params property is optional
+    template<typename Params = void, typename IB>
+    drakvuf_trap_t* register_trap(drakvuf_trap_info_t* info,
```

Comment:
Here I suggest to add comment about the ownership of returned `drakvuf_trap_t*`, i.e. that anything created using `this->register_trap` can be only disposed using `this->destroy_trap` and other methods are not allowed and cause memory leaks.

Or maybe wrap `drakvuf_trap_t` into something that is distinguishable? But idk whether this would be easy or hard to implement.

Reply:
I agree, this should be looked into.